Controllers/ToppingsController.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using PizzaOrdering_Mvc.DTO;
+using PizzaOrdering_Mvc.PizzaApp_DbContext;
+
+namespace PizzaOrdering_Mvc.Controllers
+{
+    public class ToppingsController : Controller
+    {
+        private readonly PizzaAppDbContext _pizzaAppDbContext;
+        public ToppingsController(PizzaAppDbContext pizzaAppDbContext)
+        {
+            _pizzaAppDbContext = pizzaAppDbContext;
+        }
+        public IActionResult DisplayTopping()
+        {
+            try
+            {
+                var Total_orders = _pizzaAppDbContext.Order.Count();
+                var toppings = _pizzaAppDbContext.Toppings.ToList();
+                var top_list = new List<ToppingViewModel>();
+                foreach(var top in toppings)
+                {
+                    var topping = new ToppingViewModel {
```

Comment:
We still don't need to calculate percentage for all toppings. We need to filter out most popular and least popular toppings first and then show them.